org.eclipse.xtext.generator/src/org/eclipse/xtext/generator/grammarAccess/UnicodeCharacterDatabaseNames.java

```diff
@@ -13,6 +13,8 @@
 /**
  * @author Moritz Eysholdt
  * @author Sven Efftinge
+ * 
+ * @deprecated will be removed with Xtext 2.23
  */
 @Deprecated
 public class UnicodeCharacterDatabaseNames {
```

Comment:
The deprecated impl could delegate to Character.getName(..), too such that we can remove the icu dependency earlier?